src/pin.py

```diff
@@ -52,6 +58,35 @@ def is_pin_valid():
             return False
         return True
 
+    @staticmethod
+    def read_counter():
+        try:
+            with open(login_fname, "r") as f:
+                d = json.loads(f.read())
+                Pin.counter = d["pin_counter"]
+        except:
+            Pin.counter = ATTEMPTS_MAX
+
+    @staticmethod
+    def save_counter():
+        obj = {"pin_counter": Pin.counter}
+        with open(login_fname, "w") as f:
+            f.write(json.dumps(obj))
+
+    @staticmethod
+    def reset_counter():
+        Pin.counter = ATTEMPTS_MAX
+        os.remove(login_fname)
+
+def factory_settings():
+    # unlike listdir, ilistdir is supported by unix and stm32 platform
+    files = os.ilistdir(storage_root)
+    for file in files:
+        # regular file
+        if file[1] == 0x8000:
+             f = "%s/%s" % (storage_root, file[0])
+             os.remove(f)
```

Comment:
should we try catch these? If yes, what do we do in except sentence?

Reply:
I think we should catch it and show an alert that factory reset did not work for some reason.

Also, I am thinking about folders in the `storage_root` - we have wallet files there, so if we call it a factory reset we should delete the content of the `storage_root` recursively.